### PR TITLE
redis: support mset, del, exists, touch, unlink

### PR DESCRIPTION
**Pull request by danielhochman**
Refactors support for commands which hash to multiple servers into the `FragementedRequest` abstract base and adds support for two new types of commands.

### CODE REVIEW

**Comment by mattklein123** [comment]
@ccaraman please take first pass on this.

File: test/common/redis/command_splitter_impl_test.cc [hunk 1]
```diff
@@ -382,6 +382,219 @@ TEST_F(RedisMGETCommandHandlerTest, Cancel) {
...
+  EXPECT_NE(nullptr, handle_);
+
+  RespValue expected_response;
+  expected_response.type(RespType::Integer);
+  expected_response.asInteger() = 2;
+
+  RespValuePtr response2(new RespValue());
+  response2->type(RespType::Integer);
+  response2->asInteger() = 1;
+  pool_callbacks_[1]->onResponse(std::move(response2));
```
Comment (ccaraman) [inline test/common/redis/command_splitter_impl_test.cc]: nit: do you want an expect call on the callback for this onresponse?

File: source/common/redis/command_splitter_impl.cc [hunk 2]
```diff
@@ -103,14 +116,17 @@ void MGETRequest::onChildResponse(RespValuePtr&& value, uint32_t index) {
...
+  case RespType::SimpleString: {
     pending_response_->asArray()[index].type(RespType::Error);
     pending_response_->asArray()[index].asString() = "upstream protocol error";
+    error_count_++;
     break;
   }
-  case RespType::SimpleString:
-  case RespType::BulkString:
   case RespType::Error: {
+    error_count_++;
```
Comment (mattklein123) [inline source/common/redis/command_splitter_impl.cc]: nit: add ```// fall through``` comment (or similar) on next line.

File: source/common/redis/command_splitter_impl.cc [hunk 3]
```diff
@@ -125,31 +141,154 @@ void MGETRequest::onChildResponse(RespValuePtr&& value, uint32_t index) {
...
-  for (PendingRequest& request : pending_requests_) {
-    if (request.handle_) {
-      request.handle_->cancel();
-      request.handle_ = nullptr;
+void MSETRequest::onChildResponse(RespValuePtr&& value, uint32_t index) {
+  pending_requests_[index].handle_ = nullptr;
+
+  switch (value->type()) {
+  case RespType::SimpleString: {
+    if (value->asString() == "OK")
```
Comment (mattklein123) [inline source/common/redis/command_splitter_impl.cc]: braces around all blocks

File: source/common/redis/command_splitter_impl.cc [hunk 4]
```diff
@@ -125,31 +141,154 @@ void MGETRequest::onChildResponse(RespValuePtr&& value, uint32_t index) {
...
+  RespValue single_fragment;
+  single_fragment.type(RespType::Array);
+  single_fragment.asArray().swap(values);
+
+  for (uint64_t i = 1; i < incoming_request.asArray().size(); i++) {
+    request_ptr->pending_requests_.emplace_back(*request_ptr, i - 1);
+    PendingRequest& pending_request = request_ptr->pending_requests_.back();
+
+    single_fragment.asArray()[1].asString() = incoming_request.asArray()[i].asString();
+    ENVOY_LOG(debug, "redis: parallel del: '{}'", single_fragment.toString());
```
Comment (mattklein123) [inline source/common/redis/command_splitter_impl.cc]: debug statement seems wrong

**Pull request merged**